src/AutoMapper/Mappers/ExpressionMapper.cs

```diff
@@ -247,7 +247,13 @@ private Type GetSourceType(PropertyMap propertyMap)
             {
                 if (propertyMap.SourceMember == null)
                 {
-                    return propertyMap.SourceType;
+                    return propertyMap.SourceType ??
```

Comment:
@lbargaoanu  I believe @TylerCarlson1 knows this code better - or were you just asking my advice generally?